CMakeLists.txt

```diff
@@ -174,6 +175,17 @@ find_package(Threads)
 ## SeqAn
 include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/seqan/include)
 
+## CEREAL
+if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/cereal)
+    set (CMAKE_REQUIRED_INCLUDES    ${CMAKE_INCLUDE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cereal/include)
+    include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/cereal/include)
+    message (STATUS "Cereal submodule found. Adding it to INCLUDES.")
+endif()
+check_include_file_cxx (cereal/cereal.hpp _HAVE_CEREAL)
```

Comment:
Did you check my latest push? I had already included it ^^

Reply:
In general we should think about whether we want to support building graphtyper with system libraries or only with the the shipped submodules. The way I implemented it now is that both work, but for other things like SeqAn it requires the submodule.

Reply:
> Did you check my latest push? I had already included it ^^

Nope, I was testing on an earlier commit. I will try again on the latest one.

> In general we should think about whether we want to support building graphtyper with system libraries or only with the the shipped submodules. The way I implemented it now is that both work, but for other things like SeqAn it requires the submodule.

Ok, I guess it's nice to support building with system libraries but I'm just afraid that it can be prone to bugs because we can't practically test with every submodule version and CMake library versioning is a complete mess (in my experience). At least I think that the submodules should always be preferred over system libraries... otherwise we can expect some bug reports due to someone having different submodule versions.